### PR TITLE
Set USE_NEW_WORKER_TYPE=True by default

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #8948
https://pulp.plan.io/issues/8948

### CODE REVIEW

File: CHANGES/8948.feature [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Activate the new task worker type by default.
+
+.. warning::
+
+   If you intend to stick with the old tasking system, you should configure the
+   ``USE_NEW_WORKER_TYPE`` setting to false before upgrading.
```
Comment (mdellweg) [inline CHANGES/8948.feature:6]: Didn't really know where to put this.
The changelog is part of the release notes was my thought.

File: docs/configuration/settings.rst [hunk 2]
```diff
@@ -314,8 +314,6 @@ ALLOWED_CONTENT_CHECKSUMS
 USE_NEW_WORKER_TYPE
 ^^^^^^^^^^^^^^^^^^^
 
-    .. note:: This new worker-type is still in tech-preview.
```
Comment (mdellweg) [inline docs/configuration/settings.rst:317]: IMHO we cannot make this switch _and_ say it's tech preview.

**Review by bmbouter — APPROVED** [review]
This looks great!

**Comment by mdellweg** [comment]
I need to adjust/remove the duplicated ci jobs here.

File: pulpcore/app/settings.py [hunk 3]
```diff
@@ -235,7 +235,7 @@
 
 ALLOWED_EXPORT_PATHS = []
 
-USE_NEW_WORKER_TYPE = False
+USE_NEW_WORKER_TYPE = True
```
Comment (dralley) [inline pulpcore/app/settings.py:238]: Should should rename this setting to `USE_LEGACY_WORKER_TYPE`?

Comment (mdellweg) [reply pulpcore/app/settings.py:238]: This name is used in several other repositories already. That's why i never renamed it.

File: docs/configuration/settings.rst (hunk 2 above)
Comment (bmbouter) [reply docs/configuration/settings.rst:317]: +1 to removing as tech-preview.

I think adding a  paragraph about switching from the old to the new and from the new to the old would be good. Users need guidance on the details about if they need to drain their tasks or not. Perhaps a `pulp-cli` command to show the number of tasks that need to be cancelled or maybe a `pulpcore-manager shell -c ...` command would help users know if switching is safe.

**Review by dralley — APPROVED** [review]

File: docs/configuration/settings.rst (hunk 2 above)
Comment (mdellweg) [reply docs/configuration/settings.rst:317]: cli won't work, because you first shut down the api-workers.

**Review by dralley — APPROVED** [review]

**Comment by bmbouter** [comment]
The new changes look good. @mdellweg please merge whenever the tests pass. Thank you!

File: CHANGES/8948.feature (hunk 1 above)
Comment (ipanova) [reply CHANGES/8948.feature:6]: maybe add a note in the settings.rst that the default is set to true

Comment (dralley) [reply CHANGES/8948.feature:6]: +1 to that suggestion

**Comment by mdellweg** [comment]
I do not understand the upgrade test failing here.

**Comment by bmbouter** [comment]
From what I've seen and heard the upgrade test has intermittent failures unrelated to any piece of work. Given that, I think this should merge without that passing.

**Pull request merged**